editor/editor_help_search.cpp

```diff
@@ -394,6 +419,226 @@ bool EditorHelpSearch::Runner::_is_class_disabled_by_feature_profile(const Strin
+			iterator_stack.resize(iterator_stack.size() - 1);
+		}
+
+		if (!class_doc || class_doc->name.is_empty()) {
+			return false;
+		}
+
+		// If class matches the flags, add it to the matched stack.
+		const bool class_matched =
+				(search_flags & SEARCH_CLASSES) ||
+				((search_flags & SEARCH_CONSTRUCTORS) && !class_doc->constructors.is_empty()) ||
+				((search_flags & SEARCH_METHODS) && !class_doc->methods.is_empty()) ||
+				((search_flags & SEARCH_OPERATORS) && !class_doc->operators.is_empty()) ||
+				((search_flags & SEARCH_SIGNALS) && !class_doc->signals.is_empty()) ||
+				((search_flags & SEARCH_CONSTANTS) && !class_doc->constants.is_empty()) ||
+				((search_flags & SEARCH_PROPERTIES) && !class_doc->properties.is_empty()) ||
+				((search_flags & SEARCH_THEME_ITEMS) && !class_doc->theme_properties.is_empty()) ||
+				((search_flags & SEARCH_ANNOTATIONS) && !class_doc->annotations.is_empty());
+
+		if (class_matched) {
+			matched_classes.push_back(class_doc);
+		}
+
+		// Add inheriting classes, in reverse.
+		if (class_doc && EditorHelp::get_doc_data()->inheriting.has(class_doc->name)) {
+			iterator_stack.push_back(EditorHelp::get_doc_data()->inheriting[class_doc->name].back());
+		}
+
+		return false;
+	}
+
+	if (!iterator_stack[iterator_stack.size() - 1]) {
+		iterator_stack.resize(iterator_stack.size() - 1);
+	}
+
+	return iterator_stack.is_empty();
+}
+
+bool EditorHelpSearch::Runner::_phase_fill_member_items_init() {
+	// Prepare tree.
+	class_items.clear();
+	_populate_cache();
+
+	return true;
+}
+
+TreeItem *EditorHelpSearch::Runner::_create_category_item(TreeItem *p_parent, const String &p_class, const StringName &p_icon, const String &p_metatype, const String &p_text) {
+	const String item_meta = "class_" + p_metatype + ":" + p_class;
+
+	TreeItem *item = nullptr;
+	if (_find_or_create_item(p_parent, item_meta, item)) {
+		item->set_icon(0, ui_service->get_editor_theme_icon(p_icon));
+		item->set_text(0, p_text);
+		item->set_selectable(0, false);
+		item->set_metadata(0, item_meta);
+	}
+	item->set_collapsed(true);
+
+	return item;
+}
+
+bool EditorHelpSearch::Runner::_phase_fill_member_items() {
+	if (matched_classes.is_empty()) {
+		return true;
+	}
+
+	// Pop working item from stack.
+	DocData::ClassDoc *class_doc = matched_classes[matched_classes.size() - 1];
+	matched_classes.resize(matched_classes.size() - 1);
+
+	if (class_doc) {
+		TreeItem *item = _create_class_hierarchy(class_doc, !(search_flags & SEARCH_CLASSES));
+
+		// If class has no child classes, fold it.
+		if (!item->get_first_child()) {
+			item->set_collapsed(true);
+		} else {
+			item->set_collapsed(false);
+		}
+
+		if ((search_flags & SEARCH_CLASSES)) {
+			item->clear_custom_color(0);
+			item->clear_custom_color(1);
+		} else {
+			item->set_custom_color(0, disabled_color);
+			item->set_custom_color(1, disabled_color);
+		}
+
+		// Create common header if required.
+		const bool search_all = (search_flags & SEARCH_ALL) == SEARCH_ALL;
```

Comment:
Two can play at this game.
```suggestion
		bool search_all = (search_flags & SEARCH_ALL) == SEARCH_ALL;
```

Reply:
Ditto